falcon/request.py

```diff
@@ -514,6 +522,58 @@ def cookies(self):
 
         return self._cookies.copy()
 
+    def param(self):
```

Comment:
IIRC, we decided to drop the 's' in order to avoid having to reconcile with the existing params property. However, I'm having second thoughts on this. It may be better to make a breaking change for Falcon 1.0 so that we don't carry this design cruft around for the foreseeable future. I suppose ParamProxy could implement the mapping interface to reduce the impact of this change (as well as a method to convert to a dict).

Thoughts?